src/daemon/dlt_daemon_common.c

```diff
@@ -893,19 +996,53 @@ DltDaemonContext *dlt_daemon_context_add(DltDaemon *daemon,
         }
     }
 
+#ifdef DLT_LOG_LEVEL_APP_CONFIG
+    /* configure initial log level */
+    DltDaemonContextLogSettings *settings = NULL;
+    settings = dlt_daemon_find_configured_app_id_ctx_id_settings(
+            daemon, context->apid, ctid);
+
+    if (settings != NULL) {
+        /* set log level */
+        log_level = settings->log_level;
+
+        DltDaemonContextLogSettings *ct_settings = NULL;
+        ct_settings = dlt_daemon_find_app_log_level_config(application, ctid);
+
+        /* ct_settings != null: context and app id combination already exists */
+        if (ct_settings == NULL) {
+          /* copy the configuration into the DltDaemonApplication for faster access later */
+          DltDaemonContextLogSettings *tmp =
+              realloc(application->context_log_level_settings,
+                      (++application->num_context_log_level_settings) *
+                          sizeof(DltDaemonContextLogSettings));
+          application->context_log_level_settings = tmp;
+
+          ct_settings =
+              &application->context_log_level_settings[application->num_context_log_level_settings - 1];
+          memcpy(ct_settings, settings, sizeof(DltDaemonContextLogSettings));
+          memcpy(ct_settings->ctid, ctid, DLT_ID_SIZE);
+      }
+    }
+#endif
+
     if ((strncmp(daemon->ecuid, ecu, DLT_ID_SIZE) == 0) && (daemon->force_ll_ts)) {
-        if (log_level > daemon->default_log_level)
-            log_level = daemon->default_log_level;
+#ifdef DLT_LOG_LEVEL_APP_CONFIG
+      if (log_level > daemon->default_log_level && settings == NULL)
+#else
+      if (log_level > daemon->default_log_level)
+#endif
+          log_level = daemon->default_log_level;
 
-        if (trace_status > daemon->default_trace_status)
-            trace_status = daemon->default_trace_status;
+      if (trace_status > daemon->default_trace_status)
+        trace_status = daemon->default_trace_status;
 
-        dlt_vlog(LOG_NOTICE,
-                 "Adapting ll_ts for context: %.4s:%.4s with %i %i\n",
-                 apid,
-                 ctid,
-                 log_level,
-                 trace_status);
+      dlt_vlog(LOG_NOTICE,
+               "Adapting ll_ts for context: %.4s:%.4s with %i %i\n",
+               apid,
+               ctid,
+               log_level,
+               trace_status);
     }
```

Comment:
Please check the alignment inside the whole condition "if ((strncmp(daemon->ecuid, ecu, DLT_ID_SIZE) == 0) && (daemon->force_ll_ts))"